pkg/coordinator/redemptions.go

```diff
@@ -318,3 +334,43 @@ redemptionRequestedLoop:
 
 	return result, nil
 }
+
+func EstimateRedemptionFee(
+	btcChain bitcoin.Chain,
+	redeemersOutputScripts []bitcoin.Script,
+) (int64, error) {
+	sizeEstimator := bitcoin.NewTransactionSizeEstimator().
+		// 1 P2WPKH main UTXO input.
+		AddPublicKeyHashInputs(1, true).
+		// 1 P2WPKH change output.
+		AddPublicKeyHashOutputs(1, true)
+
+	for _, script := range redeemersOutputScripts {
+		switch bitcoin.GetScriptType(script) {
+		case bitcoin.P2PKHScript:
+			sizeEstimator.AddPublicKeyHashOutputs(1, false)
+		case bitcoin.P2WPKHScript:
+			sizeEstimator.AddPublicKeyHashOutputs(1, true)
+		case bitcoin.P2SHScript:
+			sizeEstimator.AddScriptHashOutputs(1, false)
+		case bitcoin.P2WSHScript:
+			sizeEstimator.AddScriptHashOutputs(1, true)
+		default:
+			panic("non-standard redeemer output script type")
```

Comment:
I am not sure if we should trust the contract validation that much. We can have an upgrade and change the on-chain logic. I think it is OK to assume the script has to be one of the four types listed before but if it is not I would rather return an error, log it, and decide not to propose a redemption instead of killing the process. 

At some point in the future

Reply:
The main purpose of this panic is to signal that something changed on the contract side and requires our attention. I'm afraid that returning an error and just logging it will not be visible enough. WDYT?

Reply:
I think the client should panic only if there is no way out. Here there is a way out - we can decide not to propose a redemption and eventually, we can even skip the given script when proposing the redemption. Agreed about calling the attention but I would address it somewhere else, e.g. have a script type for every redemption request at tbtcscan.com

Reply:
See dc866e45